src/Modules/OrchardCore.Commerce/Abstractions/ISortableUpdaterProvider.cs

```diff
@@ -49,6 +49,26 @@ public static class SortableUpdaterProviderExtensions
         return null;
     }
 
+    /// <summary>
+    /// Selects the first provider of the type specified in <typeparamref name="TProvider"/> where
+    /// <see cref="ISortableUpdaterProvider{TModel}.IsApplicableAsync"/> evaluates to <see langword="true"/>.
+    /// </summary>
+    public static async Task<TProvider> GetFirstApplicableProviderAsync<TModel, TProvider>(
+        this IEnumerable<ISortableUpdaterProvider<TModel>> providers,
+        TModel model)
+        where TProvider : ISortableUpdaterProvider<TModel>
+    {
+        foreach (var provider in providers.OrderBy(provider => provider.Order))
+        {
+            if (provider is TProvider castedProvider && await provider.IsApplicableAsync(model))
```

Comment:
I'm not sure if this is necessary, you can just use the already existing one like this:
`await providers.GetFirstApplicableProviderAsync(model) as IProductInventoryProvider`
It can't be a problem unless `providers` contains an assortment of different types that inherit from `ISortableUpdaterProvider<T>`, but that shouldn't happen anyway.